src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Comment.cs

```diff
@@ -207,12 +207,10 @@ private void WriteCommentIndented(ReadOnlySpan<byte> utf8Value)
 
             Span<byte> output = _memory.Span;
 
-            if (_tokenType != JsonTokenType.PropertyName)
+            if (_tokenType != JsonTokenType.None)
             {
-                if (_tokenType != JsonTokenType.None)
-                {
-                    WriteNewLine(output);
-                }
+                WriteNewLine(output);
```

Comment:
please add tests (even if throws exception), at minimum:
- testing behavior you added,
- comment in the beginning of file
- comment in the end of file
- comment in the beginning of the object
- comment in the middle of the object
- comment as last element of object
- same as above but array rather than object

Reply:
Test added.